addons/drag-drop/userscript.js

```diff
@@ -43,4 +56,80 @@ export default async function ({ addon, global, console }) {
     'div[class*="selector_wrapper"]',
     'div[class*="selector_wrapper"] input[class*="action-menu_file-input"]'
   );
+
+  async function listMonitorsDroppable() {
+    while (true) {
+      const listMonitor = await addon.tab.waitForElement('div[class*="monitor_list-monitor"]', { markAsSeen: true });
+      // Get the monitor's context menu ID from the list name
+      // https://github.com/LLK/scratch-gui/blob/develop/src/components/monitor/monitor.jsx#L37
+      const contextMenuId = `monitor-${listMonitor.querySelector('div[class*="monitor_list-header"]').textContent}`;
+      droppable(listMonitor, async (files) => {
+        // Force react-contextmenu's context menu to open using their global
+        // events:
+        // https://github.com/vkbansal/react-contextmenu/blob/v2.9.4/src/ContextMenuTrigger.js#L114-L119
+        // https://github.com/vkbansal/react-contextmenu/blob/v2.9.4/src/actions.js#L27-L29
+        window.dispatchEvent(
+          new CustomEvent("REACT_CONTEXTMENU_SHOW", {
+            detail: {
+              id: contextMenuId,
+              position: { x: 0, y: 0 },
+            },
+          })
+        );
+        const contextMenu = await addon.tab.waitForElement("body > .react-contextmenu.react-contextmenu--visible");
+        // Override DOM methods to import the text file directly
+        // See: https://github.com/LLK/scratch-gui/blob/develop/src/lib/import-csv.js#L21-L22
+        const appendChild = document.body.appendChild;
+        document.body.appendChild = (fileInput) => {
+          // Restore appendChild to <body>
+          document.body.appendChild = appendChild;
+          document.body.appendChild(fileInput);
```

Comment:
Can you guarantee the next call of `appendChild` will be with the file input? Or would it be safer to only reset `appendChild` once the passed element is definitely the file input?

(Since they're a bit less intrusive—no global function overriding—I was going to suggest using a mutation observer, or possibly `waitForElement` if that's based on observers, but I'm guessing it's important here that your overrides e.g. `fileInput.click = () => {}` be set immediately.)

Reply:
I believe because [`importCSV`](https://github.com/LLK/scratch-gui/blob/develop/src/containers/monitor.jsx#L166) is called immediately (and synchronously) [on click](https://github.com/LLK/scratch-gui/blob/develop/src/components/monitor/monitor.jsx#L96), it should be guaranteed, but I can add a check just in case